tests/protocol/test_goal_state.py

```diff
@@ -20,7 +23,68 @@
 from tests.tools import AgentTestCase, patch, load_data
 
 
-class GoalStateTestCase(AgentTestCase):
+class GoalStateTestCase(AgentTestCase, HttpRequestPredicates):
+    def test_it_should_use_vm_settings_by_default(self):
```

Comment:
These tests used to be in test_wire.py, as part of the tests for WireClient.update_goal_state(). The code they are testing is now in GoalState, so I moved the tests here.

They are almost identical to its original version, except that now the test the code via GoalState/GoalState.update() instead of WireClient.update_goal_state()